### PR TITLE
Feature/timeseries dashboard

### PR DESCRIPTION
**Pull request by ngorrinm**
This pull request adds the **Mango time series dashboard**, a streamlit-based tool for loading, exploring, and forecasting time series data. Key features include multi-language support, flexible data visualization options, and built-in model training with cross-validation. The dashboard supports multiple series and models, allowing users to visualize forecasts, analyze errors, and download results for further evaluation.

### CODE REVIEW

**Comment by antoniogonzalezsuarez** [comment]
Close for the moment to avoid running tests on push

**Comment by ggsdc** [comment]
I've reopened the PR, can we solve the conflcits now that we have closed the other PR?

**Comment by ggsdc** [comment]
Please check the failing tests. Some of them are because the typings are not being imported on the test files.

File: mango_time_series/mango_time_series/dashboards/time_series_app.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+                            if "perc_err" not in data_long.columns:
+                                data_long["perc_err"] = (
+                                    data_long["err"] / data_long["y"]
+                                )
+                            if "perc_abs_err" not in data_long.columns:
+                                data_long["perc_abs_err"] = (
+                                    data_long["abs_err"] / data_long["y"]
+                                )
+
+                            # st.write(data_long)
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_app.py:270]: Delete this line

File: mango_time_series/mango_time_series/dashboards/time_series_utils/file_uploader.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
+from typing import Dict
+
+import pandas as pd
+import streamlit as st
+
+
+def preview_data(
+    uploaded_file, separator, decimal, thousands, encoding, date_format, UI_TEXT
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/file_uploader.py:8]: An argument should not be `ALL_CAPS`

File: mango_time_series/mango_time_series/dashboards/time_series_utils/file_uploader.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+    if date_columns:
+        data_frame[date_columns] = data_frame[date_columns].apply(
+            pd.to_datetime, format=date_format
+        )
+
+    st.write(data_frame.head())
+
+
+def upload_files(UI_TEXT: Dict):
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/file_uploader.py:43]: An argument should not be `ALL_CAPS`

File: mango_time_series/mango_time_series/dashboards/time_series_utils/file_uploader.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+                        "data": data_frame,
+                        "file_name": file_name,
+                    }
+            submit_button = st.form_submit_button(label=UI_TEXT["load_data"])
+        if submit_button:
+            files_loaded.update(files_loaded_tmp)
+    return files_loaded, no_model_column
+
+
+def manage_files(files_loaded, UI_TEXT):
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/file_uploader.py:150]: An argument should not be `ALL_CAPS`

File: mango_time_series/mango_time_series/dashboards/time_series_utils/forecast_template.py.j2
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/forecast_template.py.j2:1]: What type of file is this?

File: mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py [hunk 5]
```diff
@@ -0,0 +1,1008 @@
...
+
+from .data_processing import (
+    calculate_min_diff_per_window,
+)
+from ...time_series.decomposition import SeasonalityDecompose
+from ...time_series.seasonal import SeasonalityDetector
+from ...utils.utils import cast_env_to_bool
+
+
+def select_series(data: pd.DataFrame, columns: List, UI_TEXT: Dict):
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:22]: An argument should not be `ALL_CAPS`

File: mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py [hunk 6]
```diff
@@ -0,0 +1,1008 @@
...
+
+        # Filter the DataFrame by the selected value
+        filtered_data = filtered_data[filtered_data[column] == selected_value].copy()
+
+    # Create a label for the selected series
+    selected_item = {col: filtered_data.iloc[0][col] for col in columns}
+    return selected_item
+
+
+def plot_time_series(
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:49]: An argument should not be `ALL_CAPS`

File: mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py [hunk 7]
```diff
@@ -0,0 +1,1008 @@
...
+                )
+                st.plotly_chart(fig, use_container_width=True)
+
+    # "Periodogram"
+    elif select_plot == UI_TEXT["plot_options"][5]:
+        if not cast_env_to_bool("ENABLE_EXPERIMENTAL_FEATURES", default=False):
+            st.warning(UI_TEXT["experimental_features_warning"], icon="⚠️")
+            return
+        else:
+            st.info(UI_TEXT["experimental_features_info"], icon="ℹ️")
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:444]: Do we actually have emojis on a .py file?

File: mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py [hunk 8]
```diff
@@ -0,0 +1,1008 @@
...
+                title=UI_TEXT["periodogram"]["title"],
+                xaxis_title=UI_TEXT["periodogram"]["xaxis_title"],
+                yaxis_title=UI_TEXT["periodogram"]["yaxis_title"],
+                showlegend=False,
+            )
+
+            st.plotly_chart(fig, use_container_width=True)
+
+
+def setup_sidebar(time_series: pd.DataFrame, columns_id: List, UI_TEXT: Dict):
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:530]: An argument should not be `ALL_CAPS`

File: mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py [hunk 9]
```diff
@@ -0,0 +1,1008 @@
...
+                st.session_state["selected_series"] = []
+                st.rerun()
+    else:
+        st.sidebar.write(UI_TEXT["no_columns_to_filter"])
+        st.session_state["selected_series"] = [{}]
+
+    return select_agr_tmp, visualization, st.session_state["visualization_options"]
+
+
+def plot_forecast(forecast: pd.DataFrame, selected_series: List, UI_TEXT: Dict):
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:651]: An argument should not be `ALL_CAPS`

File: mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py [hunk 10]
```diff
@@ -0,0 +1,1008 @@
...
+                .replace("abs_err", "Error absoluto")
+                .replace("err", "Error")
+                .replace("weekday", "Día de la semana"),
+            )
+        )
+
+        st.plotly_chart(fig)
+
+
+def plot_error_visualization(
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:796]: An argument should not be `ALL_CAPS`

**Review by ggsdc — CHANGES_REQUESTED** [review]

**Comment by ggsdc** [comment]
Form the [PEP 8](https://peps.python.org/pep-0008/):

> [Constants](https://peps.python.org/pep-0008/#constants)
Constants are usually defined on a module level and written in all capital letters with underscores separating words. Examples include MAX_OVERFLOW and TOTAL.

I know that what you pass is actually a constant, but that does not mean that the argument is, so that is why i would have the argument be `ui_text` and when you call the function you pass it `UI_TEXT`

File: mango_time_series/mango_time_series/cli/__init__.py [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+cli_path = os.path.dirname(os.path.abspath(__file__))
+
+
+@click.group(name="mango_time_series", help="Commands in the mango_time_series cli")
+def cli():
+    pass
+
+
+# Import subcommands to register them (needs to be after cli to avoid circular imports)
+from mango_time_series.cli.dashboard import dashboard
```
Comment (ggsdc) [inline mango_time_series/mango_time_series/cli/__init__.py:14]: Why is this?

File: mango_time_series/mango_time_series/dashboards/time_series_utils/constants.py [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            "name": name,
+            "class_name": model_instance.__class__.__name__,
+            "parameters": parameters,
+        }
+    )
+    all_imports.add(model_instance.__class__.__name__)
+
+all_imports = list(all_imports)
+
+# Remove these dictionaries from here and add them to the UI_TEXT files
```
[the remark this answers is not in the document]
Comment (ngorrinm) [reply mango_time_series/mango_time_series/dashboards/time_series_utils/constants.py:57]: No, it was just a comment. Removed it.

File: mango_time_series/mango_time_series/dashboards/time_series_utils/forecast_template.py.j2
Comment (ngorrinm) [reply mango_time_series/mango_time_series/dashboards/time_series_utils/forecast_template.py.j2:1]: A jinja template

File: mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py (hunk 7 above)
Comment (ngorrinm) [reply mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:444]: Streamlit admit icons but i will change to shortcodes

Comment (ngorrinm) [reply mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:444]: https://streamlit-emoji-shortcodes-streamlit-app-gwckff.streamlit.app/

Comment (ngorrinm) [reply mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py:444]: If i do it i get this exception: StreamlitAPIException: The value "ℹ️" is not a valid emoji. Shortcodes are not allowed, please use a single character instead.

**Review by ggsdc — APPROVED** [review]

**Review by gvalleg — APPROVED** [review]

**Pull request merged**